contracts/protocols/mainnet/compound_iii/main.sol

```diff
@@ -0,0 +1,98 @@
+        marketConfig = new MarketConfig[](length);
+
+        for (uint256 i = 0; i < length; i++) {
+            positionData[i].userData = getUserData(user, markets[i]);
+            positionData[i].collateralData = getCollateralAll(user, markets[i]);
+            marketConfig[i] = getMarketConfig(markets[i]);
+        }
+    }
+
+    /**
+     *@dev get position of the user for given collateral.
+     *@notice get position details of the user in a market including overall position data, collaterals, rewards etc.
+     *@param user Address of the user whose position details are needed.
+     *@param markets Array of addresses of the market for which the user's position details are needed
+     *@param tokenIDs IDs or offsets of the token as per comet market whose collateral details are needed.
+     *@return positionData Array of overall position details of the user - balances, rewards, collaterals and flags.
+     *@return marketConfig Array of the market configuration details.
+     */
+    function getPosition(
+        address user,
+        address[] calldata markets,
+        uint8[] calldata tokenIDs
+    ) public returns (PositionData[] memory positionData, MarketConfig[] memory marketConfig) {
+        uint256 length = markets.length;
+        positionData = new PositionData[](length);
+        marketConfig = new MarketConfig[](length);
+
+        for (uint256 i = 0; i < length; i++) {
+            positionData[i].userData = getUserData(user, markets[i]);
+            positionData[i].collateralData = getAssetCollaterals(user, markets[i], tokenIDs);
+            marketConfig[i] = getMarketConfig(markets[i]);
+        }
+    }
+
+    /**
+     *@dev get market configuration.
+     *@notice returns the market stats including market supplies, balances, rates, flags for market operations,
+     *collaterals or assets active, base asset info etc.
+     *@param markets Array of addresses of the comet market for which the user's position details are needed.
+     *@return marketConfigs Array of struct containing data related to the market and the assets.
+     */
+    function getMarketConfiguration(address[] calldata markets)
+        public
+        view
+        returns (MarketConfig[] memory marketConfigs)
+    {
+        uint256 length = markets.length;
+        marketConfigs = new MarketConfig[](length);
+
+        for (uint256 i = 0; i < length; i++) {
+            marketConfigs[i] = getMarketConfig(markets[i]);
+        }
+    }
+
+    /**
+     *@dev get collaterals list where user has positiom.
+     *@notice get list of all collaterals in the market.
+     *@param user Address of the user whose collateral details are needed.
+     *@param markets Array of addresses of the comet market for which the user's collateral details are needed.
+     *@return datas array of token addresses supported in the market.
+     */
+    function getCollateralsList(address user, address[] calldata markets) public returns (address[][] memory datas) {
```

Comment:
@Richa-iitr @pradyuman-verma Do we need this function?

Reply:
Users collateral details are returned separately in `getPosition` , `getPositionAll` and the assets supported by market in `MarketConfig` so this can be removed.

Reply:
@thrilok209 Not now but let's keep it, in future if we need it somewhere while creating automation.